File: app/src/main/java/mega/privacy/android/app/providers/PasscodeFileProviderActivity.java
package mega.privacy.android.app.providers;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import mega.privacy.android.app.DatabaseHandler;
import mega.privacy.android.app.utils.PasscodeUtil;

public class PasscodeFileProviderActivity extends AppCompatActivity {

	private PasscodeUtil passcodeUtil;

	@Override
	protected void onCreate(@Nullable Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		passcodeUtil = new PasscodeUtil(this, new DatabaseHandler(this));
	}

	@Override
	protected void onPause() {
		passcodeUtil.pause();
		super.onPause();
	}
	
	@Override
	protected void onResume() {
		super.onResume();
		passcodeUtil.resume();
	}
}

File: app/src/main/java/mega/privacy/android/app/providers/PasscodeFileProviderActivity.kt
package mega.privacy.android.app.providers

import androidx.appcompat.app.AppCompatActivity
import dagger.hilt.android.AndroidEntryPoint
import mega.privacy.android.app.utils.PasscodeUtil
import javax.inject.Inject

@AndroidEntryPoint
open class PasscodeFileProviderActivity: AppCompatActivity() {

    @Inject
    lateinit var passcodeUtil: PasscodeUtil

    override fun onPause() {
        passcodeUtil.pauseUpdate()
        super.onPause()
    }

    override fun onResume() {
        super.onResume()
        passcodeUtil.resume(false)
    }
}